usdt-impl/src/common.rs

```diff
@@ -140,23 +167,17 @@ mod tests {
     #[test]
     fn test_generate_type_check() {
         let types = &[dtrace_parser::DataType::U8, dtrace_parser::DataType::String];
+        let expected = quote! {
+            {
+                fn _type_check<S: AsRef<str>>(_: u8, _: S) { }
+                let _ = || {
+                    let args = $args_lambda();
+                    _type_check(args.0, args.1);
+                };
+            }
+        };
```

Comment:
Nice, much better than string-munging :)

Reply:
ok. glad you like it. I wasn't sure if I was undoing some subtle intention...